### ISSUE TITLE
Adding a similar functionality like sklearn.make_regression, make_moons etc.

### ISSUE THREAD

**Issue by Chandu-4444** [JuliaML/MLUtils.jl#15]
I had this on my for a long time. Since I don't find any methods (Please provide the link if these already exist) for that kind of function I guess it's better to add these. I'm not sure if this is the right place to put this.

**Comment by CarloLucibello** [comment]
I think some toy dataset generators can be added, they will also be useful to document the functionality of the library, create tutorials etc...

A few synthetic datasets can be ported over from [MLDataUtils](https://github.com/JuliaML/MLDataUtils.jl/blob/master/src/datasets.jl)

**Comment by Chandu-4444** [comment]
Yeah, but the way of generating those synthetic datasets isn't convincing as the data is being loaded from a CSV file. I think these methods can be written with some more flexibility. Can I work on this?

**Comment by CarloLucibello** [comment]
> Can I work on this?

Sure, thanks. You can add them to the [generators](https://github.com/JuliaML/MLUtils.jl/blob/main/src/Datasets/generators.jl) file.

**Comment by CarloLucibello** [comment]
closing this in favour of more specific issues as we already have some datasets
